Canop/broot
# Add support for icons

Support for icons.

Included:

1) Code changes
2) Resources
3) Doc change

Not included: 

Automated testing

Manual testing steps:

1) Install `resources/icons/vscode/vscode.ttf` . For linux, copy  to ~/.local/share/fonts. You shouldn't need to run `fc-cache` but run it anyway. 

2) In a new terminal, type `echo -e "file_type_rust looks like \U001002D2"` in bash/zsh or `echo "Rust is `u{1002D2}"` in PowerShell. If you see a gears icon, good. Your setup is complete. Otherwise setup needs to be changed (I think i attached the right `ttf` ...)

3) In broot config, add `icon_theme="vscode"` 

4) Launch broot in a directory with some rust files.

I haven't spent much time mapping file extensions to file types, it was getting tiring :-). 

[image](https://raw.githubusercontent.com/Canop/broot/8b43d85d07d5f86bbaa896f74baa3f43fcb4c460/website/docs/img/icons.png)  
![icons](https://user-images.githubusercontent.com/67588957/98468006-8adbc300-21fe-11eb-9857-82bfdd9ab8b2.png)


The icon set is one of the more popular vscode icon sets- https://github.com/vscode-icons/vscode-icons/ 
That said, I couldn't figure out how to import colors in fonts.
